apps/camera/js/controllers/settings.js

```diff
@@ -304,24 +304,24 @@ SettingsController.prototype.notify = function(setting, flashDeactivated) {
   if (dontNotify) { return; }
 
   var localizeOption = setting.get('optionsLocalizable') !== false;
-  var title = this.l10nGet(setting.get('title'));
-  var optionTitle = setting.selected('title');
-  var html;
-
+  var title = '<span data-l10n-id="' + setting.get('title') + '"></span>';
   // Localize option title only if not specified in the config
-  optionTitle = localizeOption ? this.l10nGet(optionTitle) : optionTitle;
+  var optionTitle = localizeOption ? 
+    '<span data-l10n-id="' + setting.selected('title') + '"></span>' :
+    '<span>' + optionTitle + '</span>';
+  var html;
 
   // Check if the `flashMode` setting is going to be deactivated as part
   // of the change in the `hdr` setting and display a specialized
   // notification if that is the case
   if (flashDeactivated) {
     html = title + ' ' + optionTitle + '<br/>' +
-      this.l10nGet('flash-deactivated');
+      '<span data-l10n-id="flash-deactivated"></span>';
   } else {
     html = title + '<br/>' + optionTitle;
   }
 
-  this.notification.display({ text: html });
+  this.notification.display({ text: {html: html} });
```

Comment:
Hmmm... So, here's a case where notification.display takes more than l10nId:
https://github.com/mozilla-b2g/gaia/blob/master/apps/camera/js/controllers/battery.js#L114-L124

The current "text" and "attrs" should be turned into a single L10nId object {id: ..., args: ...}, but persistent and className are separate. So my approach is to turn it into:

`notification.display({
 text: l10nId,
 className: ...,
 persistent: ...
});`

and now the "text" may be a simple l10n-id:
`notification.display({
  text: 'my-string-id'
});`

or sth more complex:
`notification.display({
  text: {'id': 'my-string', args: {'name': 'John'}}
});`

it seems to me that your approach was more to do sth like this:

`notification.display({
  'id': ...,
  'args': ...,
  'persistent': ....,
  'className': ...,
});`

I'd prefer to keep L10nId object separately to keep it from potentially overlapping in a scenario like this:

1) We take your proposal
2) You add "raw" attribute next to persistent and className for some additional data
3) We want to provide "raw" variant of L10nId 
4) Attr name collision.

For that reason I'd keep text as an attribute.

Does it sound sane to you? It's your call, so if you prefer flat model, I'll write it your way.